src/couchdb/couch-pool.ts

```diff
@@ -0,0 +1,154 @@
+import { asEither, asObject, asOptional, asString } from 'cleaners'
+import nano, { ServerScope } from 'nano'
+
+/**
+ * A table of CouchDB connections,
+ * with one connection highlighted as the default.
+ */
+export interface CouchPool {
+  /** The default connection, to use for most queries. */
+  default: ServerScope
+
+  /** The cluster name being used for the default connection. */
+  defaultName: string
+
+  /** Lists the clusters that are available for connection. */
+  clusterNames: string[]
+
+  /**
+   * Grab a connection to a specific cluster.
+   * Throws if the name is missing.
+   */
+  connect: (name: string) => ServerScope
+
+  /** Grabs authentication information for a specific cluster. */
+  getCredential: (name: string) => CouchCredential | undefined
+
+  /** Grab a connection, returning undefined if the cluster is missing. */
+  maybeConnect: (name: string) => ServerScope | undefined
+}
```

Comment:
Really good API. A part of me wonders if we can do without the `maybeConnect` via static type checking. `connect` can still throw, but wouldn't ever be written to through if it constrained it's parameter. Inferring the name keys from `connectCouch` is where it gets tricky though because you inject `'default'` into the internal type, but I suppose the internal type doesn't need to be constrained by the name keys.

Reply:
The input comes from cleaner-config in the login server, so there's no type information at that point. In other situations, we have put the list of credentials in the info server.